typescript/sdk/src/hook/EvmHookModule.ts

```diff
@@ -184,29 +288,296 @@ export class EvmHookModule extends HyperlaneModule<
     return routingHookUpdates;
   }
 
+  protected async updatePausableHook({
+    currentConfig,
+    targetConfig,
+  }: {
+    currentConfig: PausableHookConfig;
+    targetConfig: PausableHookConfig;
+  }): Promise<AnnotatedEV5Transaction[]> {
+    const updateTxs = [];
+
+    if (currentConfig.paused !== targetConfig.paused) {
+      // Have to encode separately otherwise tsc will complain
+      // about being unable to infer types correctly
+      const pausableInterface = PausableHook__factory.createInterface();
+      const data = targetConfig.paused
+        ? pausableInterface.encodeFunctionData('pause')
+        : pausableInterface.encodeFunctionData('unpause');
+
+      updateTxs.push({
+        annotation: `Updating paused state to ${targetConfig.paused}`,
+        chainId: this.domainId,
+        to: this.args.addresses.deployedHook,
+        data,
+      });
+    }
+
+    return updateTxs;
+  }
+
+  protected async updateIgpHook({
+    currentConfig,
+    targetConfig,
+  }: {
+    currentConfig: IgpHookConfig;
+    targetConfig: IgpHookConfig;
+  }): Promise<AnnotatedEV5Transaction[]> {
+    const updateTxs = [];
+    const igpInterface = InterchainGasPaymaster__factory.createInterface();
+
+    // Update beneficiary if changed
+    if (!eqAddress(currentConfig.beneficiary, targetConfig.beneficiary)) {
+      updateTxs.push({
+        annotation: `Updating beneficiary from ${currentConfig.beneficiary} to ${targetConfig.beneficiary}`,
+        chainId: this.domainId,
+        to: this.args.addresses.deployedHook,
+        data: igpInterface.encodeFunctionData('setBeneficiary(address)', [
+          targetConfig.beneficiary,
+        ]),
+      });
+    }
+
+    // get gasOracleAddress using any remote domain in the current config
+    let gasOracle;
+    const domainKeys = Object.keys(currentConfig.oracleConfig);
+
+    // If possible, reuse and reconfigure the gas oracle from the first remote we know.
+    // Otherwise if there are no remotes in current config, deploy a new gas oracle with our target config.
+    // We should be reusing the same oracle for all remotes, but if not, the updateIgpRemoteGasParams step will rectify this
+    if (domainKeys.length > 0) {
+      const domainId = this.multiProvider.getDomainId(domainKeys[0]);
+      ({ gasOracle } = await InterchainGasPaymaster__factory.connect(
+        this.args.addresses.deployedHook,
+        this.multiProvider.getSignerOrProvider(this.chain),
+      )['destinationGasConfigs(uint32)'](domainId));
+
+      // update storage gas oracle
+      // Note: this will only update the gas oracle for remotes that are in the target config
+      updateTxs.push(
+        ...(await this.updateStorageGasOracle({
+          gasOracle,
+          currentOracleConfig: currentConfig.oracleConfig,
+          targetOracleConfig: targetConfig.oracleConfig,
+          targetOverhead: targetConfig.overhead, // used to log example remote gas costs
+        })),
+      );
```

Comment:
can we follow this same `updateTxs.push(...(await this.*));` pattern above in the `update()` switch-case? this would also update line 168 to `const updateTxs: AnnotatedEV5Transaction[] = [];`

Reply:
not sure I like the verbosity but ig idm too much